Makefile

```diff
@@ -27,38 +33,61 @@ else
 PLATFORM := kubernetes
 endif
 
-all: manager
+# our targets don't generate files in a way make assumes
+.PHONY: all clean test compile prepare run run_as_current_user debug deploy undeploy generate_deployment manifests fmt vet generate docker-build docker-push controller-gen help
+
+all: compile
+
+### clean: deletes the potential leftover generated files from the previous build runs
+clean:
+	@rm -Rf $(TEMP_DIR)
 
 ### test: Run tests
 test: generate fmt vet manifests
-	go test ./... -coverprofile cover.out
+	go test -v -coverprofile cover.out ./...
 
-### manager: Build manager binary
-manager: generate fmt vet
+### compile: Build the manager binary
+compile: generate fmt vet
 	go build -o bin/manager main.go
 
-### run: Run against the configured Kubernetes cluster in ~/.kube/config
-run: generate fmt vet manifests
+### prepare: Prepares the cluster for running the operator - deploys CRDs, rbac and the service account
+prepare: _mk_temp manifests
+	$(eval PREPAPRE_OUTPUT_DIR := $(shell mktemp -p $(TEMP_DIR) -d))
```

Comment:
maybe it's better to move the long command to a separate bash script?
Motivation - bash script format is simpler than Makefile format.

Reply:
That script would need to rely on 4 parameters (or env vars) and would be 5 lines.. That's why I opted for this 5-lines of make target. I will add some comments to explain what's going on there.